src/Bicep.Core/Syntax/AstVisitor.cs

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax
+{
+    /// <summary>
+    /// Visits an <see href="https://en.wikipedia.org/wiki/Abstract_syntax_tree">abstract syntax tree (AST)</see>.
```

Comment:
>AST [](http://example.com/codeflow?start=103&length=3)

Is it worth adding a comment somewhere that explains what our definition of AST is in terms of our classes?